app/controllers/authorizer.rb

```diff
@@ -0,0 +1,56 @@
+module Authorizer
+  def tapp_admin
+    is_admin(ENV['TAPP_ADMINS'])
+  end
+
+  def cp_admin
+    is_admin(ENV['CP_ADMINS'])
+  end
+
+  def correct_applicant
+    if ENV['RAILS_ENV'] == 'production'
+      if get_utorid != get_applicant(params)
+        render status: 403, file: 'public/403.html'
+      end
+    end
+  end
+
+  private
+  def is_admin(admins)
+    admins = admins.split(',')
+    if ENV['RAILS_ENV'] == 'production'
+      if !admins.include?(get_utorid)
+        render status: 403, file: 'public/403.html'
+      end
+    end
+  end
+
+  def get_applicant(params)
+    offer = Offer.find_by(link: params[:mangled])
+    if offer
+      offer = offer.format
+      return offer[:applicant][:utorid]
+    else
+      render status: 404, file: 'public/404.html'
+    end
+  end
+
+  def get_utorid
```

Comment:
this depends on shib enable reverse proxy stuffing in request headers when it forwards